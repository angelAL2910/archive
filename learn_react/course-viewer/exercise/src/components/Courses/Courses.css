.add{
    margin-top: 5%;
}

.add-course{
    color: white;
    background-color: #007bff;
    margin: 3% 0;
    border-radius: 5px;
}

.add-course a{
    color: white;
}

.table-hover a{
    color: red;
}

.title a{
    color: #007bff;
}