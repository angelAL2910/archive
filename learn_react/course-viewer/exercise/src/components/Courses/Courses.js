import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

import {
  addEditableCourse,
  loadCourses,
  loadAuthors
} from "../../actions/actionCreators";
import CourseItem from "../CourseItem/CourseItem";
import "../../App.css";
import "./Courses.css";

export class Courses extends React.Component {
  constructor(props) {
    super(props);
    this.textInput = React.createRef();
  }

  componentDidMount() {
    if (this.props.courses.length == 0) this.props.addCoursestoStore();
    if (this.props.authors.length == 0) this.props.addAuthorsToStore();
  }

  render() {
    let courses = this.props.courses;
    return (
      <div className="element">
        <h2>Courses</h2>
        <button className="btn add-course">
          <Link to="/course">Add Course</Link>
        </button>
        <table className="table table-hover">
          <thead>
            <tr>
              <th />
              <th>Title</th>
              <th>Author</th>
              <th>Category</th>
              <th />
            </tr>
          </thead>
          <tbody>
            {courses.map((course, i) => (
              <CourseItem key={course.id} course={course} {...this.props} />
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}

export function mapStateToProps({ courses, authors }) {
  return {
    courses,
    authors
  };
}

export const mapDispatchToProps = dispatch => ({
  addCoursestoStore: () => {
    dispatch(loadCourses());
  },

  addAuthorsToStore: () => {
    dispatch(loadAuthors());
  },

  addCourseToStoreForEditing: course => {
    dispatch(addEditableCourse(course));
  }
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Courses);
