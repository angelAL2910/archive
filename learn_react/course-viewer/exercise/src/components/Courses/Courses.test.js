import React from "react";
import renderer from "react-test-renderer";
import { mapStateToProps, mapDispatchToProps, Courses } from "./Courses";
import { BrowserRouter } from "react-router-dom";

jest.mock('../CourseItem/CourseItem', () => 'CourseItem')

const fakeStore = {
  courses: [
    {
      id: 1,
      title: "Securing React Apps with Auth01",
      slug: "Securing-React-Apps-with-Auth01",
      authorId: "2",
      category: "JavaScript"
    }
  ],
  authors: [
    {
      id: 1,
      name: "Cory House"
    }
  ],
  editCourse: [
    {
      id: 1,
      title: "Securing React Apps with Auth01",
      slug: "Securing-React-Apps-with-Auth01",
      authorId: "2",
      category: "JavaScript"
    }
  ],
  routing: {
    locationBeforeTransitions: null
  },
};

test("Testing the courses page", () => {
  expect(mapStateToProps(fakeStore).authors.length).toEqual(1);
  const dispatch = jest.fn();

   mapDispatchToProps(dispatch).addCoursestoStore();
   expect(dispatch.mock.calls[0][0]).toEqual({ type: 'LOAD_COURSES'});

});

test("Testing the courses page", () => {
    expect(mapStateToProps(fakeStore).authors.length).toEqual(1);
    const dispatch = jest.fn();
     mapDispatchToProps(dispatch).addAuthorsToStore();
     expect(dispatch.mock.calls[0][0]).toEqual({ type: 'LOAD_AUTHORS'});

  });

  test("Testing the courses page", () => {
    let newCourse = {
      id: 2,
      title: "Securing React Apps with Auth012",
      slug: "Securing-React-Apps-with-Auth012",
      authorId: "2",
      category: "JavaScript2"
    };
    expect(mapStateToProps(fakeStore).authors.length).toEqual(1);
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).addCourseToStoreForEditing(newCourse);
    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'ADD_EDITABLE_COURSE', data: newCourse});

  });


 test("Protecting against regression issues.....", () => {
   let tree = renderer.create(
     <BrowserRouter>
       <Courses courses={fakeStore.courses} authors={fakeStore.authors} />
     </BrowserRouter>
  ).toJSON();

   expect(tree).toMatchSnapshot();

 });
