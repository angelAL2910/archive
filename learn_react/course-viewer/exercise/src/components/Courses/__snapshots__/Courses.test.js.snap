// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Protecting against regression issues..... 1`] = `
<div
  className="element"
>
  <h2>
    Courses
  </h2>
  <button
    className="btn add-course"
  >
    <a
      href="/course"
      onClick={[Function]}
    >
      Add Course
    </a>
  </button>
  <table
    className="table table-hover"
  >
    <thead>
      <tr>
        <th />
        <th>
          Title
        </th>
        <th>
          Author
        </th>
        <th>
          Category
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <CourseItem
        authors={
          Array [
            Object {
              "id": 1,
              "name": "Cory House",
            },
          ]
        }
        course={
          Object {
            "authorId": "2",
            "category": "JavaScript",
            "id": 1,
            "slug": "Securing-React-Apps-with-Auth01",
            "title": "Securing React Apps with Auth01",
          }
        }
        courses={
          Array [
            Object {
              "authorId": "2",
              "category": "JavaScript",
              "id": 1,
              "slug": "Securing-React-Apps-with-Auth01",
              "title": "Securing React Apps with Auth01",
            },
          ]
        }
      />
    </tbody>
  </table>
</div>
`;
